### PR TITLE
ENH: add np.nanprod

### PR DESCRIPTION
**Pull request by shoyer**
This PR adds an implementation of `nanprod`.

The actual function is a two-liner adapted from `nansum`. Most of this PR
consists of documentation and tests (for which I took the opportunity to do
some consolidation).

A method with the same functionality exists in pandas, and I was surprised to
discover that it's not in numpy.

### CODE REVIEW

File: numpy/lib/nanfunctions.py [hunk 1]
```diff
@@ -510,6 +513,76 @@ def nansum(a, axis=None, dtype=None, out=None, keepdims=0):
...
+    y : ndarray or numpy scalar
+
+    See Also
+    --------
+    numpy.prod : Product across array propagating NaNs.
+    isnan : Show which elements are NaN.
+
+    Notes
+    -----
+    Numpy integer arithmetic is modular. If the size of a sum exceeds the
```
Comment (charris) [inline numpy/lib/nanfunctions.py]: sum -> product

File: numpy/lib/tests/test_nanfunctions.py [hunk 2]
```diff
@@ -260,119 +265,7 @@ def test_nanstd(self):
...
-        res = np.nansum(mat)
-        assert_(np.isscalar(res))
-
-
-class TestNanFunctions_MeanVarStd(TestCase):
-
-    nanfuncs = [np.nanmean, np.nanvar, np.nanstd]
-    stdfuncs = [np.mean, np.var, np.std]
-
+class SharedNanFunctionsTestsMixin(object):
```
Comment (charris) [inline numpy/lib/tests/test_nanfunctions.py]: Interesting approach.

**Comment by charris** [comment]
LGTM modulo noun nitpick.

File: numpy/lib/nanfunctions.py (hunk 1 above)
Comment (shoyer) [inline numpy/lib/nanfunctions.py]: fixed

**Comment by charris** [comment]
Merged, thanks.

**Pull request merged**